File: client-sdk/csharp/sdk/src/VertexCache/Sdk/Protocol/Parsers/ErrorResponseParser.cs
using VertexCache.Sdk.Results;

namespace VertexCache.Sdk.Protocol.Parsers
{
    public class ErrorResponseParser : IResponseParser
    {
        public bool CanParse(string response) => response.StartsWith("-");

        public VCacheResult Parse(string response) =>
            VCacheResult.FailureWithRaw(VCacheErrorCode.ServerError, response, response);
    }
}

File: client-sdk/go/sdk/protocol/parsers/error_response_parser.go
package parsers

import (
	"strings"

	"vertexcache/sdk/results"
)

type ErrorResponseParser struct{}

func (p *ErrorResponseParser) CanParse(line string) bool {
	return strings.HasPrefix(line, "-")
}

func (p *ErrorResponseParser) Parse(line string) *results.VCacheResult {
	return results.NewErrorWithRaw(results.ServerError, line, line)
}
